src/dbnode/storage/database.go

```diff
@@ -340,6 +358,32 @@ func (d *db) addNamespacesWithLock(namespaces []namespace.Metadata) error {
 	return nil
 }
 
+func (d *db) updateNamespaceSchemasWithLock(schemaUpdates []namespace.Metadata) error {
+	for _, n := range schemaUpdates {
+		// ensure namespace exists
+		existing, ok := d.namespaces.Get(n.ID())
+		if !ok { // should never happen
+			return fmt.Errorf("non-existing namespace marked for schema update: %v", n.ID().String())
+		}
+		currentId := "none"
+		current, found := existing.SchemaRegistry().GetLatest()
+		if found {
+			currentId = current.DeployId()
+		}
+		// log schema update
+		latest, found := n.Options().SchemaRegistry().GetLatest()
+		if !found {
```

Comment:
In what situation could this occur? Should this be an `invariant violated` type of situation? Either a comment explaining how this could happen or making the error message more descriptive would be helpful

Reply:
@Haijuncao Any update on this?

Reply:
will add more comment.

I think this may happen when the schema registry for a namespace is accidentally wiped out by an admin. an empty registry is technically a valid registry, so we want to catch it here. We can also catch it later in `curNamespace.SetSchemaRegistry` (where it checks lineage).  But the error message may be not as clear.